mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -276,6 +277,177 @@ def surrogate_predict(self, configurations: pd.DataFrame, context: Optional[pd.D
         return ret
 
     def acquisition_function(self, configurations: pd.DataFrame,
-                             context: Optional[pd.DataFrame] = None) -> Callable:
+                             context: Optional[pd.DataFrame] = None) -> npt.NDArray:
         # This seems actually non-trivial to get out of skopt, so maybe we actually shouldn't implement this.
         raise NotImplementedError()
+
+
+class SmacOptimizer(BaseBayesianOptimizer):
+    """ Wrapper class for SMAC based Bayesian optimization.
+
+    Parameters
+    ----------
+    parameter_space : ConfigSpace.ConfigurationSpace
+        The parameter space to optimize.
+    """
+    def __init__(
+        self,
```

Comment:
maybe consider `def __init__(self, *, parameter_space, ...)` to enforce named parameters? (not sure how it will work with the Factory, though)

Reply:
Maybe open a separate PR for this later?

Reply:
I don't think that will work out well with the factory method.  I think it's ok to leave it as is for now.  We could tackle changing that behavior in a separate PR.